benchmarks/python/core.py

```diff
@@ -259,40 +260,107 @@ def run_benchmark(
         outputs: Output of the target function
     """
 
-    assert device in [
+    # Check that the device is `cuda` or `host:{compile/steady/dynamic}`.
+    assert device.split(":")[0] in [
         "cuda",
         "host",
-    ], f"Unsupported device type: {device}. Use one of cuda/host."
+    ], f'Unsupported device type: {device.split(":")[0]}. Use one of cuda/host.'
 
-    if device == "host":
+    host_bench_mode = None
+
+    # Store warmup rounds locally to modify for host:steady/dynamic cases.
+    global warmup_rounds
```

Comment:
Why does this need to be `global`?

Reply:
This was a remnant from some code changes I was trying.
Removed it now.